lib/grape/middleware/base.rb

```diff
@@ -21,8 +22,38 @@ def call(env)
       def call!(env)
         @env = env
         before
+
         @app_response = @app.call(@env)
-        after || @app_response
+
+        # Merge Headers from API into @header
```

Comment:
Probably not, I trust you to dig this a bit and only do it if it makes any sense :)